tests/unit/amo/components/TestAddon.js

```diff
@@ -1566,4 +1568,42 @@ describe(__filename, () => {
       expect(installButton).toHaveProp('uninstall', uninstall);
     });
   });
+
+  it('displays a notice to the admin/developer on a disabled addon', () => {
```

Comment:
There should be one test case for an admin and one test case for the developer of the add-on.

`is_addon_developer` does not control anything because the API performs the permissions check. We should probably leave a comment mentioning that: "non-public add-ons require an account listed as a developer of the add-on or admin rights".